_posts/2024/2024-02-06-about-caching-in-eo.md

```diff
@@ -0,0 +1,235 @@
+---
+layout: post
+date: 2024-02-06
+title: "Build cache in EO and other build systems"
+author: Alekseeva Yana
+---
+
+
+## Introduction 
+In [EO](https://github.com/objectionary/eo), caching is used to speed up program compilation.
+Recently we found a caching 
+[bug](https://github.com/objectionary/eo/issues/2790) in `eo-maven-plugin`
+for EO version `0.34.0`. The error occurred because the algorithm compared
+the compilation time and caching time to search for the needed file.
+This is not the most reliable verification method,
+because caching time does not have to be equal to compilation time.
+That we came to the conclusion that we need caching with a reliable verification method
+that does not require reading a file system.
+
+The goal of this blog is to research caching in frequently used build systems (`ccache`, `Maven`, `Gradle`)
+and to create effective caching in [EO](https://github.com/objectionary/eo).
+
+<!--more-->
+
+## Build caching of existing build systems
+
+### ccache/sccache
+In compiled programming languages, building a project takes a long time.
```

Comment:
@Yanich96 "long time" ? How much is it? I build all my projects relatively fast.